libs/clients/regulations/src/lib/regulations.ts

```diff
@@ -0,0 +1,134 @@
+import { Inject } from '@nestjs/common'
+import { RESTDataSource, RequestOptions } from 'apollo-datasource-rest'
+import { DataSourceConfig } from 'apollo-datasource'
+import {
+  ISODate,
+  RegQueryName,
+  Regulation,
+  RegulationLawChapterTree,
+  RegulationListItem,
+  RegulationMinistryList,
+  RegulationRedirect,
+  RegulationSearchResults,
+  RegulationYears,
+} from './regulations.types'
+
+export const REGULATIONS_OPTIONS = 'REGULATIONS_OPTIONS'
+
+export interface RegulationsServiceOptions {
+  url: string
+  ttl?: number
+}
+
+export class RegulationsService extends RESTDataSource {
+  constructor(
+    @Inject(REGULATIONS_OPTIONS)
+    private readonly options: RegulationsServiceOptions,
+  ) {
+    super()
+    this.baseURL = `${this.options.url}`
+    this.initialize({} as DataSourceConfig<any>)
+  }
+
+  willSendRequest(request: RequestOptions) {
+    request.headers.set('Content-Type', 'application/json')
+  }
+  /*
+  Example api routes for regulation
+  regulation/[name]/current
+  regulation/[name]/original
+  regulation/[name]/diff
+  regulation/[name]/d/[date]
+  regulation/[name]/d/[date]/diff
+  regulation/[name]/d/[date]/diff/[earlierDate]
+*/
+  async getRegulation(
+    viewType: 'current' | 'diff' | 'original' | 'd',
+    name: RegQueryName,
+    date?: ISODate,
+    isCustomDiff?: boolean,
+    earlierDate?: ISODate | 'original',
+  ): Promise<Regulation | RegulationRedirect | null> {
+    let params: string = viewType
+
+    if (viewType === 'd') {
+      if (date) {
+        params = 'd/' + date
+        if (isCustomDiff) {
+          params += '/diff' + (earlierDate ? '/' + earlierDate : '')
+        }
+      } else {
+        // Treat `viewType` 'd' with no `date` as 'current'
+        // ...either that or throwing an error...
+        // ...or tightening the type signature to prevent that happening.
+        params = 'current'
+      }
+    }
+    const response = await this.get<Regulation | RegulationRedirect | null>(
+      `/regulation/${name}/${params}`,
+      {
+        cacheOptions: { ttl: this.options.ttl ?? 600 }, // defaults to 10 minutes
+      },
+    )
+    return response
+  }
+
+  async getRegulations(
+    type: 'newest',
+    page?: number,
+  ): Promise<RegulationSearchResults | null> {
+    page = page && page > 1 ? page : undefined
+    const response = await this.get<RegulationSearchResults | null>(
+      `regulations/${type}${page ? '?page=' + page : ''}`,
+      {
+        cacheOptions: { ttl: this.options.ttl ?? 600 }, // defaults to 10 minutes
+      },
+    )
+    return response
+  }
+
+  async getRegulationsSearch(
+    q?: string,
+    rn?: string,
+    year?: string,
+    ch?: string,
+  ): Promise<RegulationListItem[] | null> {
+    const response = await this.get<RegulationListItem[] | null>(
+      `search`,
+      { q, rn, year, ch },
+      {
+        cacheOptions: { ttl: 1 /* this.options.ttl ?? 600*/ }, // defaults to 10 minutes
```

Comment:
Should this perhaps be?
```suggestion
        cacheOptions: { ttl: this.options.ttl ?? 600}, // defaults to 10 minutes
```

Reply:
fixed